libtenzir/src/node.cpp

```diff
@@ -284,6 +284,48 @@ auto node_state::get_endpoint_handler(const http_request_description& desc)
   return result->second;
 }
 
+auto spawn_components(node_actor::stateful_pointer<node_state> self) -> void {
+  using component_plugin_map = std::unordered_map<std::string, const component_plugin*>;
+  component_plugin_map loaded = {};
+  component_plugin_map not_loaded = {};
+  for (const auto* component : plugins::get<component_plugin>()) {
+    not_loaded.emplace(component->component_name(), component);
+  }
+  auto load_recursive = [&](auto load_recursive, const std::string& needed) {
+    if (loaded.contains(needed)) {
+      return;
+    }
```

Comment:
The variable name `needed` is confusing. We should rename it to `named`.
```suggestion
  auto load_recursive = [&](auto load_recursive, const std::string& needed) {
    if (loaded.contains(needed)) {
      return;
    }
```